server/leaderboard_rank_cache_test.go

```diff
@@ -61,17 +63,19 @@ func TestLocalLeaderboardRankCache_Insert_Descending(t *testing.T) {
 	u4 := uuid.Must(uuid.NewV4())
 	u5 := uuid.Must(uuid.NewV4())
 
-	cache.Insert("lid", 0, LeaderboardSortOrderDescending, u3, 33, 34)
-	cache.Insert("lid", 0, LeaderboardSortOrderDescending, u2, 22, 23)
-	cache.Insert("lid", 0, LeaderboardSortOrderDescending, u4, 44, 45)
-	cache.Insert("lid", 0, LeaderboardSortOrderDescending, u1, 11, 12)
-	cache.Insert("lid", 0, LeaderboardSortOrderDescending, u5, 55, 56)
-
-	assert.EqualValues(t, 1, cache.Get("lid", 0, u5))
-	assert.EqualValues(t, 2, cache.Get("lid", 0, u4))
-	assert.EqualValues(t, 3, cache.Get("lid", 0, u3))
-	assert.EqualValues(t, 4, cache.Get("lid", 0, u2))
-	assert.EqualValues(t, 5, cache.Get("lid", 0, u1))
+	order := LeaderboardSortOrderDescending
+
+	cache.Insert("lid", order, 33, 34, nil, nil, 0, u3)
+	cache.Insert("lid", order, 22, 23, nil, nil, 0, u2)
+	cache.Insert("lid", order, 44, 45, nil, nil, 0, u4)
+	cache.Insert("lid", order, 11, 12, nil, nil, 0, u1)
+	cache.Insert("lid", order, 55, 56, nil, nil, 0, u5)
+
+	assert.EqualValues(t, 1, cache.Get("lid", order, 55, 56, 0, u5))
+	assert.EqualValues(t, 2, cache.Get("lid", order, 44, 45, 0, u4))
+	assert.EqualValues(t, 3, cache.Get("lid", order, 33, 34, 0, u3))
+	assert.EqualValues(t, 4, cache.Get("lid", order, 22, 23, 0, u2))
+	assert.EqualValues(t, 5, cache.Get("lid", order, 11, 12, 0, u1))
```

Comment:
Should we also add a test for tied scores that are untied by the subscore?

Reply:
Good idea! Will do :+1: